hal/src/nRF52840/openthread/alarm.c

```diff
@@ -279,7 +282,7 @@ static void GetOffsetAndCounter(uint32_t *aOffset, uint32_t *aCounter)
 
 static uint64_t GetTime(uint32_t aOffset, uint32_t aCounter, AlarmIndex aIndex)
 {
-    uint64_t result = (uint64_t)aOffset * US_PER_OVERFLOW + TicksToTime(aCounter, kUsTimer);
+    uint64_t result = (uint64_t)aOffset * US_PER_OVERFLOW + (DWT->CYCCNT - sSystemMillisClock) / SYSTEM_US_TICKS; 
```

Comment:
I am a bit on the fence about this. I'd rather keep OT-specific functionality untouched and use `DWT` for higher precision only for our `hal_timer_micros()`.

Reply:
Sounds good! Fixed.